packages/next-swc/crates/next-api/src/app.rs

```diff
@@ -487,15 +486,19 @@ impl AppEndpoint {
     async fn output(self: Vc<Self>) -> Result<Vc<AppEndpointOutput>> {
         let this = self.await?;
 
-        let (app_entry, ty) = match this.ty {
-            AppEndpointType::Page { ty: _, loader_tree } => {
-                (self.app_page_entry(loader_tree), "page")
-            }
+        let (app_entry, ty, ssr_and_client) = match this.ty {
+            AppEndpointType::Page { ty, loader_tree } => (
+                self.app_page_entry(loader_tree),
+                "page",
+                matches!(ty, AppPageEndpointType::Html),
```

Comment:
We only need to use SSR and Client side modules for the Html endpoint and not for the Rsc endpoint